Mathlib/RingTheory/IntegralRestrict.lean

```diff
@@ -0,0 +1,129 @@
+/-
+Copyright (c) 2023 Andrew Yang, Patrick Lutz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.RingTheory.IntegrallyClosed
+import Mathlib.RingTheory.Norm
+import Mathlib.RingTheory.DedekindDomain.IntegralClosure
+/-!
+# Restriction of various maps between fields to integrally closed subrings.
+
+## Main definition
+- `galRestrict`: The restriction `Aut(L/K) → Aut(B/A)` as an `MulEquiv` in an ALKB setup.
+
+## TODO
+Define the restriction of norms and traces.
```

Comment:
Concerning the norm we already have [RingOfIntegers.norm](https://leanprover-community.github.io/mathlib4_docs/Mathlib/NumberTheory/NumberField/Norm.html#RingOfIntegers.norm). Or do you mean something else?

Reply:
Yes but I find the AKLB setup (as in this file) easier to work with and I plan to redefine RingOfIntegers.norm using the new definition.